pilot/pkg/model/context.go

```diff
@@ -610,25 +610,27 @@ func (node *Proxy) GetInterceptionMode() TrafficInterceptionMode {
 // GetUniqueSuffixes Return a slice containing the strings with the longesest
 // unique suffixes
 func GetUniqueSuffixes(stringSlice []string) []string {
-	out := []string{}
+	domainMap := map[string]struct{}{}
 
 	// Iterate through the slice finding longest strings with unique suffixes
-	for _, stringOne := range stringSlice {
-		workString := ""
-		for _, stringTwo := range stringSlice {
-			// If strings have same suffix
-			if strings.HasSuffix(stringOne, stringTwo) {
-				// Keep the longest string from the first range
-				if len(stringOne) > len(stringTwo) {
-					workString = stringOne
+	for i := 0; i < len(stringSlice); i++ {
+		domainMap[stringSlice[i]] = struct{}{}
+		for j := 0; j < len(stringSlice); j++ {
+			if j != i {
+				// If strings have same suffix
+				if strings.HasSuffix(stringSlice[j], stringSlice[i]) {
+					delete(domainMap, stringSlice[i])
+					domainMap[stringSlice[j]] = struct{}{}
 				}
 			}
 		}
+	}
 
-		// Append first range working string if a new one was found
-		if workString != "" {
-			out = append(out, workString)
-		}
+	out := make([]string, 0, len(domainMap))
+
+	for domain := range domainMap {
+		out = append(out, domain)
```

Comment:
you need to sort the output for a deterministic list of domains. Map iteration is non deterministic